sycl/include/CL/sycl/detail/cg.hpp

```diff
@@ -422,15 +444,18 @@ class CGFill : public CG {
   vector_class<char> MPattern;
   Requirement *MPtr;
 
-  CGFill(vector_class<char> Pattern, void *Ptr,
-         vector_class<vector_class<char>> ArgsStorage,
-         vector_class<detail::AccessorImplPtr> AccStorage,
-         vector_class<shared_ptr_class<const void>> SharedPtrStorage,
-         vector_class<Requirement *> Requirements,
-         vector_class<detail::EventImplPtr> Events)
+  CGFill(std::vector<char> Pattern, void *Ptr,
+         std::vector<std::vector<char>> ArgsStorage,
```

Comment:
GIT merge did that automatically - does it need to be changed to vector_class? If so, there should be more instances

Reply:
@tovinkere all public interfaces should use SYCL-style containers to maintain ABI compatibility across different libstd implementations. This particular header file is accessible from user code (through `handler.hpp`), thus requires usage of `vector_class`. At the moment it is not clear, what's public and what's not. You can consult #1136 that moves internal files to where they belong to. Also, it is a good idea to use SYCL classes on along the call chain (like, `queue::submit`, `queue_impl::submit` and `queue_impl::submit_impl`) to avoid unnecessary conversions.

Reply:
Updated CG and derived class constructors to reflect the current types that merge failed to resolve.

Reply:
@alexbatashev  Can you elaborate on the idea of using SYCL classes along the call chain?

Reply:
@tovinkere let's say you have some class `A`. It has a method `foo` that accepts a `vector_class`. In our implementation most of the spec-defined classes do nothing but call corresponding methods from impl classes. So, our `A_impl::foo` must also accept a `vector_class` to avoid unnecessary conversions. And you can't change `vector_class` to `std::vector` inside user-accessible header files (that is everything inside `include/CL/sycl/**`) to avoid unwanted side effects of different class layout across standard library implementations. Instead you keep passing it as `vector_class`. You can do the conversion in the library itself (`source/**`) whenever you need, though. On the other hand, internal stuff, like program manager, does not interact with user directly, thus it can safely use `std::` containers (for performance, usability, maintainability, portability, and other numerous reasons we use standard library). It is still a good idea to avoid type conversion in your code.